test/shared/src/main/scala/zio/test/Gen.scala

```diff
@@ -143,8 +143,26 @@ object Gen extends GenZIO with FunctionVariants {
   /**
    * A generator of strings. Shrinks towards the empty string.
    */
-  final val anyString: Gen[Random with Sized, String] =
-    Gen.string(Gen.anyChar)
+  final def anyString: Gen[Random with Sized, String] =
+    Gen.string(Gen.anyUnicodeChar)
+
+  /**
+   * A generator of Unicode characters. Shrinks toward '0'.
+   * Heavily inspired from:
+   * https://github.com/typelevel/scalacheck/blob/ab15a9fe012ca7c9feb48ee188e4d167df76d6ba/src/main/scala/org/scalacheck/Arbitrary.scala#L122-L134
+   */
+  final val anyUnicodeChar: Gen[Random, Char] = {
+    def unicodeRange(min: Int, max: Int): Gen[Random, Char] =
+      fromEffectSample {
+        // 0xFFFF and 0xFFFE are not characters in the Unicode standard.
+        // See http://www.unicode.org/charts/PDF/UFFF0.pdf
+        nextInt(max - min + 1)
+          .map(r => (min + r).toChar)
+          .map(Sample.shrinkIntegral(0))
```

Comment:
Be careful about shrinking logic here. We want to make sure that shrinks obey the invariants of the generator. Here we would shrink to `0` even if the minimum value in the range is not equal to `0`. It doesn't matter in this case because this is a `private` method and it is being called with a `0` minimum value but just a heads up.

Reply:
ah that makes sense, I should use `min` instead here. Will update 👍